### PR TITLE
Add iter:task_vma iterators detection

### PR DESCRIPTION
**Pull request by Rtoax**
kernel commit 3a7b35b899de("bpf: Introduce task_vma bpf_iter") support bpf_iter__task_vma.

It also adds standard API to check the iter probes presence in code:
```
  BPFfeature::has_prog_iter_task_vma
```
Example:
```
  # bpftrace -e 'iter:task_vma {printf("%s` %d %lx-%lx\n", comm, pid, ctx->vma->vm_start, ctx->vma->vm_end);}'
  Attaching 1 probe...
  bpftrace 119480 55b92c380000-55b92c386000
  bpftrace 119480 55b92c386000-55b92c391000
  bpftrace 119480 55b92c391000-55b92c397000
  ...
  bpftrace 119480 7ffd55d87000-7ffd55da8000
  bpftrace 119480 7ffd55dde000-7ffd55de2000
  bpftrace 119480 7ffd55de2000-7ffd55de4000
```

### CODE REVIEW

File: tests/data/btf_data.h [hunk 1]
```diff
@@ -58,10 +58,16 @@
 //    struct file *file;
 //  };
 //
+//  struct bpf_iter__task_vma {
+//    struct task_struct *task;
+//    struct vm_area_struct *vma;
+//  };
+//
```
Comment (viktormalik) [inline tests/data/btf_data.h:65]: Updating the comment is not sufficient, you have to re-generate the BTF data. See below for commands.

Comment (Rtoax) [reply tests/data/btf_data.h:70]: Finish it, thanks.

```bash
 $ cat a.c                                                                   
 [... the above definitions ...]                                             
 $ gcc -o btf -g btf.c                                                       
 $ pahole -J btf   # v1.17 and higher                                        
 $ objcopy --dump-section .BTF=data btf                                      
 $ xxd -i data > data.h
```

**Comment by viktormalik** [comment]
Please fix the clang-format errors.

**Comment by Rtoax** [comment]
> Please fix the clang-format errors.

Fixed with `git clang-format origin/master`, thanks.

**Review by danobi — APPROVED** [review]
lgtm

**Comment by danobi** [comment]
Oh, needs changelog entry as well, please

**Comment by Rtoax** [comment]
> Oh, needs changelog entry as well, please

Added changelog entry, thanks.

**Pull request merged**